### PR TITLE
Deduplicate search POI route_track, enable *.travel.obf search, add binary tests, add route types, transfer FindByRenderingTypesRules class from tools

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2277

File: OsmAnd/src/net/osmand/plus/resources/ResourceManager.java [hunk 1]
```diff
@@ -1493,13 +1493,13 @@ public BinaryMapIndexReader[] getTransportRoutingMapFiles() {
...
 
-	public BinaryMapIndexReader[] getQuickSearchFiles() {
+	public BinaryMapIndexReader[] getQuickSearchFiles(boolean includeTravelObf) {
 		Collection<BinaryMapReaderResource> fileReaders = getFileReaders();
 		List<BinaryMapIndexReader> readers = new ArrayList<>(fileReaders.size());
 		for (BinaryMapReaderResource r : fileReaders) {
 			BinaryMapIndexReader shallowReader = r.getShallowReader();
 			if (shallowReader != null && (shallowReader.containsPoiData() || shallowReader.containsAddressData()) &&
-					!r.getFileName().endsWith(IndexConstants.BINARY_TRAVEL_GUIDE_MAP_INDEX_EXT)) {
+					(includeTravelObf || !r.getFileName().endsWith(IndexConstants.BINARY_TRAVEL_GUIDE_MAP_INDEX_EXT))) {
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1502]: We can probably pass something like List<String> ignoredExtensions for this check

Comment (RZR-UA) [reply OsmAnd/src/net/osmand/plus/resources/ResourceManager.java]: refactored

File: OsmAnd-java/src/main/java/net/osmand/osm/MapPoiTypes.java [hunk 2]
```diff
@@ -547,6 +547,9 @@ public void initFromInputStream(InputStream is) {
                                 }
                             }
                         }
+						case "poi_types" -> {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/osm/MapPoiTypes.java:550]: Formatting

Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/osm/MapPoiTypes.java]: fixed but there is a lot of more bad formatting...

File: OsmAnd-java/src/main/java/net/osmand/render/RenderingRuleSearchRequest.java [hunk 3]
```diff
@@ -348,4 +377,70 @@ public String toString() {
...
\ No newline at end of file
+
+	@Nullable
+	public String searchIconByTags(@NotNull Map<String, String> tags) {
+		return searchBestPropertyByTransformedTags(tags, RenderingRulesStorage.POINT_RULES,
+				storage.PROPS.R_ICON, storage.PROPS.R_ICON_ORDER);
+	}
+
+	@Nullable
+	private String searchBestPropertyByTransformedTags(@NotNull Map<String, String> transformedTags, int rulesNumber,
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/render/RenderingRuleSearchRequest.java:388]: why do we need transformed tags? is it wrong name?

Comment (vshcherb) [reply OsmAnd-java/src/main/java/net/osmand/render/RenderingRuleSearchRequest.java:388]: what does it mean best? all comparatives in the code best / better is a bad approach. method should directly state what's sorting about, first rule or max tags matching

File: OsmAnd-java/src/main/java/net/osmand/render/RenderingRulesStorage.java [hunk 4]
```diff
@@ -616,26 +617,7 @@ public static void main(String[] args) throws XmlPullParserException, IOExceptio
...
-							renderingConstants.put(parser.getAttributeValue("", "name"), 
-									parser.getAttributeValue("", "value"));
-						}
-					}
-				}
-			}
-		} finally {
-			defaultIS.close();
-		}
+		final Map<String, String> renderingConstants = readRenderingConstantsFromInputStream(defaultIS);
```
Comment (ivanPyrohivskyi) [inline OsmAnd-java/src/main/java/net/osmand/render/RenderingRulesStorage.java:620]: ok, refactoring here

File: OsmAnd-java/src/main/java/net/osmand/render/RenderingRulesStorage.java [hunk 5]
```diff
@@ -701,11 +683,66 @@ protected static void testSearch(RenderingRulesStorage storage) {
...
+				RenderingRulesStorage depends = new RenderingRulesStorage(name, null);
+				depends.parseRulesFromXmlInputStream(RenderingRulesStorage.class.getResourceAsStream(name + ".render.xml"), ref, false);
+				return depends;
+			}
+		};
+		storage.parseRulesFromXmlInputStream(is, resolver, false);
+		return storage;
+	}
+
+	public static RenderingRulesStorage initWithStylesFromResources(@Nullable String... resourceFileNames) {
```
Comment (ivanPyrohivskyi) [inline OsmAnd-java/src/main/java/net/osmand/render/RenderingRulesStorage.java:704]: ok

File: OsmAnd-java/src/main/java/net/osmand/search/SearchUICore.java [hunk 6]
```diff
@@ -289,6 +289,13 @@ public boolean sameSearchResult(SearchResult r1, SearchResult r2) {
 									|| (subType1.startsWith("route_hiking_") && subType1.endsWith("n_poi"))) {
 								similarityRadius = 50000;
 							}
+							final String ROUTE_ID = "route_id";
```
Comment (ivanPyrohivskyi) [inline OsmAnd-java/src/main/java/net/osmand/search/SearchUICore.java:292]: ok, deduplicate

File: OsmAnd-java/src/main/java/net/osmand/render/RenderingRuleSearchRequest.java (hunk 3 above)
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/render/RenderingRuleSearchRequest.java:388]: searchTopOrderedPropertyByTags

**Pull request merged**